src/WAC.php

```diff
@@ -130,6 +164,46 @@ private function getUserGrants($resourcePath, $webId) {
 		return $grants;
 	}
 
+	private function getOriginGrants($resourcePath, $origin) {
+		$aclPath = $this->getAclPath($resourcePath);
+		if (!$aclPath) {
+			return array();
+		}
+		$acl = $this->filesystem->read($aclPath);
+
+		$graph = new \EasyRdf_Graph();
+		$graph->parse($acl, Format::TURTLE, $_SERVER['REQUEST_SCHEME'] . "://" . $_SERVER['SERVER_NAME']);
+
+		// error_log("GET GRANTS for $origin");
+
+		$grants = $this->getPublicGrants($resourcePath);
+
+		$matching = $graph->resourcesMatching('http://www.w3.org/ns/auth/acl#origin');
+		//error_log("MATCHING " . sizeof($matching));
+		// Find all grants machting our origin;
+		foreach ($matching as $match) {
+			$grantedOrigin = $match->get("<http://www.w3.org/ns/auth/acl#origin>");
+			if ($grantedOrigin == $origin) {
+				$accessTo = $match->get("<http://www.w3.org/ns/auth/acl#accessTo>");
+				//error_log("$origin accessTo $accessTo");
+				$default = $match->get("<http://www.w3.org/ns/auth/acl#default>");
+				$modes = $match->all("<http://www.w3.org/ns/auth/acl#mode>");
+				if ($default) {
```

Comment:
Not sure if `if - foreach - if - foreach` is better or `foreach - if - if`... Since this is already nested in a foreach...  and/or maybe use `yield` to speed things up? :thinking:

Not sure...